netcore/System.Private.CoreLib/src/System/Array.Mono.cs

```diff
@@ -228,10 +228,10 @@ public static unsafe Array CreateInstance (Type elementType, int length)
 			if (length < 0)
 				ThrowHelper.ThrowLengthArgumentOutOfRange_ArgumentOutOfRange_NeedNonNegNum ();
 
-			RuntimeType? t = elementType.UnderlyingSystemType as RuntimeType;
-			if (t == null)
+			RuntimeType? runtimeType = elementType.UnderlyingSystemType as RuntimeType;
+			if (runtimeType == null)
 				ThrowHelper.ThrowArgumentException (ExceptionResource.Arg_MustBeType, ExceptionArgument.elementType);
-			return InternalCreate (t, 1, &length, null);
+			return InternalCreate (runtimeType._impl.Value, 1, &length, null);
 		}
```

Comment:
Note that this will work right now, but will not work with unloadable types later as there will be nothing to keep the type alive during the icall. This is why the coreclr seems to pass type objects to icalls instead of pointers.

Reply:
Ok, I guess `is_subclass_of` icall also should be improved then

Reply:
Well, that is easily solvable by GC.KeepAlive. Also, there is probably already reference on the stack. The return handle may be a problem though.

Reply:
@vargaz so will GC.KeepAlive work for this case?

Reply:
@EgorBo It would have to be after the InternalCreate call. Note that `GC.KeepAlive` is no-op, all it does is to ensure the lifetime of an object until the `GC.KeepAlive` call is made.

Reply:
hm... it was defined in `icall-def-netcore.h` but it's not an icall actually..

Reply:
It doesn't matter right now, will have to be audited later.